crates/democracy/src/lib.rs

```diff
@@ -0,0 +1,1154 @@
+        /// Invalid preimage
+        PreimageInvalid,
+        /// No proposals waiting
+        NoneWaiting,
+        /// The given account did not vote on the referendum.
+        NotVoter,
+        /// Too high a balance was provided that the account cannot afford.
+        InsufficientFunds,
+        /// Invalid upper bound.
+        WrongUpperBound,
+        /// Maximum number of votes reached.
+        MaxVotesReached,
+        /// Maximum number of proposals reached.
+        TooManyProposals,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        /// Weight: see `begin_block`
+        fn on_initialize(n: T::BlockNumber) -> Weight {
+            Self::begin_block(n).unwrap_or_else(|e| {
+                sp_runtime::print(e);
+                0
+            })
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Propose a sensitive action to be taken.
+        ///
+        /// The dispatch origin of this call must be _Signed_ and the sender must
+        /// have funds to cover the deposit.
+        ///
+        /// - `proposal_hash`: The hash of the proposal preimage.
+        /// - `value`: The amount of deposit (must be at least `MinimumDeposit`).
+        ///
+        /// Emits `Proposed`.
+        ///
+        /// Weight: `O(p)`
+        #[pallet::weight(T::WeightInfo::propose())]
+        pub fn propose(
+            origin: OriginFor<T>,
+            proposal_hash: T::Hash,
+            #[pallet::compact] value: BalanceOf<T>,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+            ensure!(value >= T::MinimumDeposit::get(), Error::<T>::ValueLow);
+
+            let index = Self::public_prop_count();
+            let real_prop_count = PublicProps::<T>::decode_len().unwrap_or(0) as u32;
+            let max_proposals = T::MaxProposals::get();
+            ensure!(real_prop_count < max_proposals, Error::<T>::TooManyProposals);
+
+            T::Currency::reserve(&who, value)?;
+            PublicPropCount::<T>::put(index + 1);
+            <DepositOf<T>>::insert(index, (&[&who][..], value));
+
+            <PublicProps<T>>::append((index, proposal_hash, who));
+
+            Self::deposit_event(Event::<T>::Proposed(index, value));
+            Ok(())
+        }
+
+        /// Signals agreement with a particular proposal.
+        ///
+        /// The dispatch origin of this call must be _Signed_ and the sender
+        /// must have funds to cover the deposit, equal to the original deposit.
+        ///
+        /// - `proposal`: The index of the proposal to second.
+        /// - `seconds_upper_bound`: an upper bound on the current number of seconds on this proposal. Extrinsic is
+        ///   weighted according to this value with no refund.
+        ///
+        /// Weight: `O(S)` where S is the number of seconds a proposal already has.
+        #[pallet::weight(T::WeightInfo::second(*seconds_upper_bound))]
```

Comment:
This is quite an unfortunate effect of storing the deposits in a vec. Since this is paritys code I'll assume this is for a good reason, but I haven't come across it yet